Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidArrayListCodeFixProvider.cs

```diff
@@ -47,24 +48,80 @@ private async Task<Document> SwapType(Document document, SyntaxToken token)
 		{
 			SyntaxNode root = await document.GetSyntaxRootAsync();
 
-			var variable = token.Parent?.Parent as VariableDeclarationSyntax;
-			var type = variable?.Type;
-
-			if (root != null && type != null)
+			if (token.Parent?.Parent is VariableDeclarationSyntax variable)
 			{
-				var list = SyntaxFactory.ParseTypeName("System.Collections.Generic.List<int>");
-				
-				root = root.ReplaceNode(type, list.WithTriviaFrom(type).WithAdditionalAnnotations(Formatter.Annotation));
+				var type = variable?.Type;
+				root = ReplaceTypeWithList(root, type);
+				var replacedVariable = root.GetAnnotatedNodes(annotation).FirstOrDefault()?.Ancestors()
+					.OfType<VariableDeclarationSyntax>().FirstOrDefault();
+				if (
+					replacedVariable != null &&
+					replacedVariable.Variables[0]?.Initializer?.Value is ObjectCreationExpressionSyntax creation)
+				{
+					root = ReplaceTypeWithList(root, creation.Type);
+				}
 
 				return document.WithSyntaxRoot(root);
 			}
 
 			return document;
 		}
 
-		public sealed override FixAllProvider GetFixAllProvider()
+		private SyntaxNode ReplaceTypeWithList(SyntaxNode root, TypeSyntax existingType)
+		{
+			if (root != null && existingType != null)
+			{
+				var parameterType = SyntaxFactory.ParseTypeName("int")
+					.WithAdditionalAnnotations(RenameAnnotation.Create(), annotation);
+				var list = CreateGenericTypeSyntax("List", parameterType);
+
+				root = root.ReplaceNode(existingType,
+					list.WithTriviaFrom(existingType).WithAdditionalAnnotations(Formatter.Annotation));
+			}
+
+			return root;
+		}
+
+		public override FixAllProvider GetFixAllProvider()
+		{
+			return null;
```

Comment:
I suspect that we don't need this override - the default behavior is probably to not have a multi-fix behavior.

Reply:
There is an Analyzer that prevents you from leaving this method unatended